### PR TITLE
More code cleanups

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
"Typical" cleanup, `final var` here and there, a few code simplifications and reformatting, `@Override`...

### CODE REVIEW

**Review by maehne — APPROVED** [review]
Looks good to me!

**Comment by MarcinOrlowski** [comment]
For now, this PR won't be green unless #791 is landed

File: src/main/java/com/cburch/logisim/analyze/gui/KarnaughMapPanel.java [hunk 1]
```diff
@@ -218,7 +219,7 @@ public void focusLost(FocusEvent e) {
 
   private void computePreferredSize() {
     selInfo = null;
-    Graphics2D g = (Graphics2D) getGraphics();
+    final var g = (Graphics2D) getGraphics();
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/analyze/gui/KarnaughMapPanel.java:222]: `g` -> `gfx`?

File: src/main/java/com/cburch/logisim/analyze/model/Assignments.java [hunk 2]
```diff
@@ -34,8 +34,6 @@
 class Assignments {
   private final Map<String, Boolean> map = new HashMap<>();
 
-  public Assignments() {}
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/analyze/model/Assignments.java:37]: Why you remove this constructor and left similar ones in place as dummy in other places?

**Review by maehne — APPROVED** [review]
Looks good to me. I made some comments related to your renaming of variables named `g` to `gfx`, which you did in some places, but not in others. You may consider to make it even more consistent.

File: src/main/java/com/cburch/draw/tools/AbstractTool.java [hunk 3]
```diff
@@ -54,10 +54,14 @@
...
   /** This is because a popup menu may result from the subsequent mouse release. */
   @Override
-  public void cancelMousePress(Canvas canvas) {}
+  public void cancelMousePress(Canvas canvas) {
+    // dummy
+  }
 
   @Override
-  public void draw(Canvas canvas, Graphics g) {}
+  public void draw(Canvas canvas, Graphics g) {
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/draw/tools/AbstractTool.java:62]: Fixed

File: src/main/java/com/cburch/draw/tools/LineTool.java [hunk 4]
```diff
@@ -62,23 +61,21 @@ public LineTool(DrawingAttributeSet attrs) {
...
-      }
+      return (Math.abs(my - py) < Math.abs(mx - px))
+          ? Location.create(mx, py)
+          : Location.create(px, my);
     }
     return Location.create(mx, my); // should never happen
   }
 
   @Override
   public void draw(Canvas canvas, Graphics g) {
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/draw/tools/LineTool.java:75]: Fixed

File: src/main/java/com/cburch/logisim/analyze/gui/KarnaughMapPanel.java (hunk 1 above)
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/analyze/gui/KarnaughMapPanel.java:222]: Fixed

File: src/main/java/com/cburch/logisim/analyze/model/Assignments.java (hunk 2 above)
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/analyze/model/Assignments.java:37]: This one is `public` and the only constructor, so can be removed. The other places I marked with `// dummy` were private constructors.

**Comment by MarcinOrlowski** [comment]
@BFH-ktt1 it's green now

**Review by maehne — APPROVED** [review]
Looks good to me! Thanks @MarcinOrlowski for taking into account my feedback!

**Pull request merged**